### PR TITLE
Add support for `gems.rb`/`gems.locked` during `bundle install`.

### PR DESCRIPTION
**Pull request by ioquatix**
Fixes part of #52 - doesn't work with `gems.rb`.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -213,14 +213,26 @@ async function installBundler(bundlerVersionInput, platform, rubyPrefix, engine,
...
+  }
+
+  if (await bundleInstallSpecific('Gemfile', 'Gemfile.lock', platform, engine, version)) {
+    return true
+  }
+
+  console.log('No Gemfile/gems.rb, skipping "bundle install" and caching')
+}
+
+async function bundleInstallSpecific(gemsPath, lockPath, platform, engine, version) {
```
Comment (eregon) [inline index.js:227]: `gemsPath` is a confusing name, how about `gemfilePath`?

**Review by eregon — COMMENTED** [review]
It looks like you missed this case:
https://github.com/ruby/setup-ruby/blob/a2ee959e496fb0bbfd6c8f8b64b2f049238cfc56/index.js#L178

**Comment by eregon** [comment]
Thanks for the PR.
Please let me review before merging since I'm the maintainer.

**Comment by eregon** [comment]
Is there any reason to use `gems.rb` over `Gemfile`?
Any adoption of that? It seems extremely few projects use it.

I'm not looking to forward to support every possible experimental Bundler feature, due to the cost to maintain that.
It seems Rails also considers it as a rather pointless change: https://github.com/rails/rails/issues/31295

File: index.js [hunk 2]
```diff
@@ -213,14 +213,26 @@ async function installBundler(bundlerVersionInput, platform, rubyPrefix, engine,
 }
 
 async function bundleInstall(platform, engine, version) {
-  if (!fs.existsSync('Gemfile')) {
-    console.log('No Gemfile, skipping "bundle install" and caching')
-    return
+  if (await bundleInstallSpecific('gems.rb', 'gems.locked', platform, engine, version)) {
```
Comment (eregon) [inline index.js:216]: `bundleInstallSpecific` does not `return` in the successful case.
Does the `if` work here? It doesn't seem so:
```
> async function foo() { console.log("foo"); }
> async function call() { var r=await foo(); console.log(r); if (r) { console.log("OK") } else { console.log("NOPE") } }
undefined
> call()
foo
Promise { <pending> }
> undefined
NOPE
```

**Comment by eregon** [comment]
I'll revert this, let's review & discuss it properly before merging.

**Comment by ioquatix** [comment]
I’m already using it so please don’t revert it. Why don’t we just fix the issues you have. I tested both cases in my own repos and it seemed to work fine.

**Comment by ioquatix** [comment]
I am using master.

https://github.com/ioquatix/bake/runs/1064255388?check_suite_focus=true

**Comment by eregon** [comment]
I see, sorry about that.
Using `@master` is expected to be unstable: https://github.com/actions/toolkit/blob/master/docs/action-versioning.md#compatibility

**Comment by ioquatix** [comment]
Sure, I completely understand your POV, but I think it warranted just a tiny bit more discussion, given how easy it is to make the changes you requested.

**Comment by ioquatix** [comment]
https://github.com/ruby/setup-ruby/pull/84

**Pull request merged**